dashboard/app/controllers/activities_controller.rb

```diff
@@ -24,7 +24,7 @@ def milestone
 
     if params[:script_level_id]
       @script_level = ScriptLevel.cache_find(params[:script_level_id].to_i)
-      @level = @script_level.level
+      @level = (params[:level_id] && Level.find(params[:level_id].to_i)) || @script_level.levels[0]
```

Comment:
I switched to `oldest_active_level` here, but once this change is rolled out milestone posts will always include a level_id param. This is just a fallback in case I missed something, and to cover people who post progress right after the push.